### PR TITLE
Nonpr 1717 upgrades

### PR DESCRIPTION
**Pull request by Leeroyc720**
**Acceptance-tests results:** 

- 24 Scenarios (24 passed)
- 196 Steps (196 passed)
- Runtime: 4m1.908s

**Tests Results:** 

- [info] Run completed in 11 seconds, 86 milliseconds.
- [info] Total number of tests run: 53
- [info] Suites: completed 13, aborted 0
- [info] Tests: succeeded 53, failed 0, canceled 0, ignored 0, pending 0
- [info] All tests passed.
- [info] Passed: Total 53, Failed 0, Errors 0, Passed 53

### CODE REVIEW

File: .bsp/sbt.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"name":"sbt","version":"1.4.9","bspVersion":"2.0.0-M5","languages":["scala"],"argv":["/Applications/IntelliJ IDEA CE.app/Contents/jbr/Contents/Home/bin/java","-Xms100m","-Xmx100m","-classpath","/Users/leeroychiweshe/Library/Application Support/JetBrains/IdeaIC2020.2/plugins/Scala/launcher/sbt-launch.jar","xsbt.boot.Boot","-bsp"]}
```
Comment (craigtd) [inline .bsp/sbt.json:1]: @Leeroyc720 this file should not be included as part of the CR. Recommend you remove this and then ensure it is added correctly to your .gitignore file

File: .gitignore [hunk 2]
```diff
@@ -31,3 +31,4 @@ test-result
 tmp
 .idea_modules
 bin
+/.bsp/
```
Comment (craigtd) [inline .gitignore:34]: Doesn't look like this worked as planned. Propose follow this pattern: https://github.com/hmrc/nonrep-attachment-processor/blob/master/.gitignore#L16

**Review by craigtd — CHANGES_REQUESTED** [review]
Just a change to .gitignore required and remove .bsp dir from the PR

**Comment by craigtd** [comment]
There are some deprecation warnings - we can address these via a separate ticket @Leeroyc720

**Comment by craigtd** [comment]
Confirmed all unit tests were successful and manually tested the frontend using service manager dependencies.

File: .bsp/sbt.json (hunk 1 above)
Comment (Leeroyc720) [reply .bsp/sbt.json]: Thanks for spotting this, its now resolved

**Review by craigtd — APPROVED** [review]
Looks fine.

File: build.sbt [hunk 3]
```diff
@@ -16,7 +16,7 @@ lazy val scoverageSettings = {
 lazy val compile = Seq(
   ws,
   "uk.gov.hmrc" %% "bootstrap-frontend-play-27" % "2.24.0",
-  "uk.gov.hmrc" %% "play-ui" % "8.21.0-play-27",
+  "uk.gov.hmrc" %% "play-ui" % "9.2.0-play-27",
   "uk.gov.hmrc" %% "govuk-template" % "5.63.0-play-27",
   "uk.gov.hmrc" %% "auth-client" % "3.0.0-play-27",
   "com.typesafe.play" %% "play-json-joda" % "2.9.0",
```
Comment (greg-gajda) [inline build.sbt:22]: There are more recent versions available for govuk-template, govuk-template and play-json-joda.
Is there any reason why we don;t update them?

File: build.sbt [hunk 4]
```diff
@@ -25,7 +25,7 @@ lazy val compile = Seq(
 )
 
 def test(scope: String) = Seq(
-  "uk.gov.hmrc" %% "hmrctest" % "3.9.0-play-26" % scope,
```
Comment (greg-gajda) [inline build.sbt:28]: It looks like play-27 is used for application and play-26 for unit tests.
Shouldn't that be improved as well?

File: build.sbt [hunk 5]
```diff
@@ -25,7 +25,7 @@ lazy val compile = Seq(
...
+  "uk.gov.hmrc" %% "hmrctest" % "3.10.0-play-26" % scope,
   "org.scalatest" %% "scalatest" % "3.0.8" % scope,
```
Comment (greg-gajda) [inline build.sbt:29]: 3.2.9 is possibly the latest stable tag for scalatest.

File: project/build.properties [hunk 6]
```diff
@@ -1 +1 @@
-sbt.version=1.3.13
```
Comment (greg-gajda) [inline project/build.properties:1]: Why 1.4.9 instead of https://github.com/sbt/sbt/releases/tag/v1.5.2 ?

Comment (Leeroyc720) [reply project/build.properties:1]: Hi Greg, for most of these upgrades I was using these suggestions: https://catalogue.tax.service.gov.uk/service/nrs-retrieval-frontend

File: build.sbt (hunk 3 above)
Comment (Leeroyc720) [reply build.sbt:22]: I will  bump the play-ui version up, do you happen to have a link for the play-json-joda page as some of these dependencies don't get listed on the 'what's running where page?'

File: build.sbt (hunk 4 above)
Comment (Leeroyc720) [reply build.sbt:28]: That is the latest version that has been suggested, don't think they have play-27 versions for hmrctest released yet, been going through most repos it seems this the current latest that's being suggested to use.

**Pull request merged**